src/components/OrientationAwareIcon.tsx

```diff
@@ -17,32 +18,43 @@ export const OrientationAwareIcon = (props: Props) => {
   const { orientation, dimensions } = useContext(OrientationContext);
   const {
     Icon,
+    iconName,
     iconStyle,
     landscapeIconStyle,
     landscapeStyle,
     size = normalize(24),
     style
   } = props;
 
+  const SelectedIcon = Icon || IconComponent.NamedIcon;
+  const isNamedIcon = SelectedIcon === IconComponent.NamedIcon;
+
   const needLandscapeStyle =
     orientation === 'landscape' || dimensions.width > consts.DIMENSIONS.FULL_SCREEN_MAX_WIDTH;
 
-  // we need adjustments only on iOS, so otherwise just return the item with the usual props
-  // TODO: is this still needed with Expo 48?
-  if (!needLandscapeStyle || !(device.platform === 'ios')) {
-    return <Icon {...props} />;
-  }
+  const shouldAdjust = !(device.platform === 'ios') || !needLandscapeStyle;
```

Comment:
is it still needed only for ios to make adjustments with our latest version?

Reply:
Yes, this check is still necessary. It is also clearly visible in the screenshots.

|with|without|
|--|--|
<img width="425" alt="image" src="https://github.com/user-attachments/assets/beac6436-8d14-49ba-9cda-66ceea87c9bf" />|<img width="429" alt="image" src="https://github.com/user-attachments/assets/5573d261-f7b1-4912-84cc-d07b4bb8dd01" />

Reply:
hmm...so android always need some adjustments (portrait and landscape) and ios just portrait?

a small request: `!(device.platform === 'ios')` should become `!device.platform !== 'ios'`

Reply:
Why do you want this? I tried to keep this control the same as before. What do you mean by `!device.platform !== 'ios'`?

Reply:
i mean `device.platform !== 'ios'`, sorry. i find it strange to have the `!` before a `===`

Reply:
Wouldn't it be more understandable if `device.platform === 'android'`? If it's not iOS, it's Android. 🤷‍♂️

Reply:
haha, yes! i mean, it could be web but we do not use it. so yes, letzt go for the check on android